File: src/main/java/org/stt/StopWatch.java
package org.stt;

import java.util.logging.Logger;

public class StopWatch {
    private static final Logger LOG = Logger.getLogger(StopWatch.class.getName());
    private final String name;
    private final long start;

    public StopWatch(String name) {
        this.name = name;
        start = System.currentTimeMillis();
    }

    public void stop() {
        LOG.finest(() -> String.format("%s : %d", name, (System.currentTimeMillis() - start)));
    }
}

File: src/main/kotlin/org/stt/StopWatch.kt
package org.stt

import java.util.logging.Logger

class StopWatch(private val name: String) {
    private val start: Long

    init {
        start = System.currentTimeMillis()
    }

    fun stop() = LOG.finest { String.format("%s : %d", name, System.currentTimeMillis() - start) }

    companion object {
        private val LOG = Logger.getLogger(StopWatch::class.java.name)
    }
}
